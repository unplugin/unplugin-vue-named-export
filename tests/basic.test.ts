import path from 'node:path'
import { describe } from 'vitest'
import {
  RollupEsbuildPlugin,
  RollupRemoveVueFilePathPlugin,
  RollupVue,
  rollupBuild,
  testFixtures,
} from '@vue-macros/test-utils'
import VueNamedExport from '../src/vite'

describe('fixtures', async () => {
  await testFixtures(
    ['tests/fixtures/*.vue'],
    (args, id) =>
      rollupBuild(id, [
        RollupVue(),
        VueNamedExport(),
        RollupRemoveVueFilePathPlugin(),
        RollupEsbuildPlugin({
          target: 'esnext',
        }),
      ]),
    { cwd: path.resolve(__dirname, '..'), promise: true },
  )
})
