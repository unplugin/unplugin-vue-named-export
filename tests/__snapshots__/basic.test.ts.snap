// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/ElCard.vue 1`] = `
"// ElCard.js
import { defineComponent, createElementBlock, openBlock, renderSlot } from 'vue';

"use strict";
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ElCard",
  setup(__props) {
    console.log("ElCard");
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});

const ElCard = _sfc_main;

export { ElCard, ElCard as default };
"
`;

exports[`fixtures > tests/fixtures/UPPER-CASE.vue 1`] = `
"// UPPER-CASE.js
import { defineComponent } from 'vue';

"use strict";
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "UPPER-CASE",
  setup(__props) {
    console.log("UPPER-CASE");
    return () => {
    };
  }
});

const UpperCase = _sfc_main;

export { UpperCase, UpperCase as default };
"
`;

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"// basic.js
import { defineComponent, createElementBlock, openBlock, renderSlot } from 'vue';

"use strict";
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "basic",
  setup(__props) {
    console.log("basic");
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});

const Basic = _sfc_main;

export { Basic, Basic as default };
"
`;

exports[`fixtures > tests/fixtures/el-button.vue 1`] = `
"// el-button.js
import { defineComponent, createElementBlock, openBlock, renderSlot } from 'vue';

"use strict";
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "el-button",
  setup(__props) {
    console.log("el-button");
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});

const ElButton = _sfc_main;

export { ElButton, ElButton as default };
"
`;

exports[`fixtures > tests/fixtures/my_button.vue 1`] = `
"// my_button.js
import { defineComponent, createElementBlock, openBlock, renderSlot } from 'vue';

"use strict";
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "my_button",
  setup(__props) {
    console.log("my_button");
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});

const MyButton = _sfc_main;

export { MyButton, MyButton as default };
"
`;
